ampad-rnaseq-de-clean/rosmap_cohort.R

```diff
@@ -1,56 +1,76 @@
+                                config::get("sequencing metadata")$version) %>%
+  full_join(get_data(config::get("synID mapping")$synID), by = c("sample" = "id")) %>%
+  select(specimenID, everything(), -sample, -versionNumber) %>%
+  distinct()
+
+# Remove one version of duplicate counts "syn4212581", "syn4212582"
+counts <- get_data(config::get("counts")$synID)[-c(1:4),] %>%
+  dplyr::select(-syn4212582) %>%
+  tibble::column_to_rownames(var = "feature") %>%
+  t() %>%
+  as.data.frame() %>%
+  tibble::rownames_to_column(var = "sample") %>%
+  left_join(get_data(config::get("synID mapping")$synID), by = c("sample" = "id")) %>%
+  select(-sample, -versionNumber) %>%
+  tibble::column_to_rownames(var = "specimenID") %>%
+  t()
 
 # Filter for required mapping columns and metadata variables, keep distinct pairs
-data$thefile[data$object == "mapping"][[1]] <- distinct(dplyr::select(data$thefile[data$object == "mapping"][[1]], projid, rnaseq_id))
-data$thefile[data$object == "clinical metadata"][[1]] <-dplyr::select(data$thefile[data$object == "clinical metadata"][[1]], -age_first_ad_dx, -age_death, -age_at_visit_max)
+mapping <- get_data(config::get("mapping")$synID, config::get("mapping")$version) %>%
+  dplyr::select(., projid, rnaseq_id) %>%
+  distinct()
+clinical_metadata <- get_data(config::get("clinical metadata")$synID, config::get("clinical metadata")$version) %>%
+  dplyr::select(-age_first_ad_dx, -age_death, -age_at_visit_max)
 
 # Join metadata files
-metadata <- data$thefile[data$object == "assay metadata"][[1]] %>%
-  dplyr::left_join(data$thefile[data$object == "sequencing metadata"][[1]], by = c("Sampleid" = "sample")) %>% 
-  dplyr::select(-projid) %>% 
-  dplyr::left_join(data$thefile[data$object == "mapping"][[1]], by = c("Sampleid" = "rnaseq_id")) %>% 
-  dplyr::left_join(data$thefile[data$object == "clinical metadata"][[1]], by = c("projid")) %>% 
-  dplyr::left_join(data$thefile[data$object == "clinical metadata - uncensored ages"][[1]], by = c("projid"))
-  
-# Pick higher quality RIN batch
-metadata <- metadata %>% 
-  dplyr::group_by(Sampleid) %>%
-  dplyr::top_n(1, RINcontinuous)
+metadata <- get_data(config::get("assay metadata")$synID,
+                     config::get("assay metadata")$version) %>%
+  dplyr::left_join(sequencing_metadata,
+                   by = c("Sampleid" = "specimenID")) %>%
+  dplyr::select(-projid) %>%
+  dplyr::left_join(mapping,
+                   by = c("Sampleid" = "rnaseq_id")) %>%
+  dplyr::left_join(clinical_metadata,
+                   by = c("projid")) %>%
+  dplyr::left_join(get_data(config::get("protected clinical metadata")$synID,
+                            config::get("protected clinical metadata")$version),
+                   by = c("projid"))
 
 #######################
 # Data pre-processing #
 #######################
 
 # Identify variables with missing data
-missingid <- metadata$Sampleid[is.na(metadata$cogdx)|
+removed_samples <- metadata$Sampleid[is.na(metadata$cogdx)|
```

Comment:
The column names are also potentially confounding application. Should we specify a string of relevant column names and make this more portable to any user input?